code/game/objects/structures/tables_racks.dm

```diff
@@ -333,8 +335,12 @@
 /obj/structure/table/proc/table_destroy(var/destroy_type, var/mob/user)
 
 	if(destroy_type == TBL_DESTROY)
-		new framestack(src.loc)
-		new buildstack(src.loc)
+		var/frameamount = framestack
+		for(var/i = 1, i <= framestackamount, i++)
+			new frameamount(get_turf(src))
+		var/buildamount = buildstack
+		for(var/i = 1, i <= buildstackamount, i++)
+			new buildamount(get_turf(src))

```

Comment:
why not just `new framestack(get_turf(src))` and `new buildstack(get_turf(src))` directly ?